src/rgw/rgw_lc.cc

```diff
@@ -516,20 +516,19 @@ class LCObjsLister {
 
   bool get_obj(rgw_bucket_dir_entry *obj) {
     if (obj_iter == objs.end()) {
-      delay();
-      return false;
-    }
-    if (is_truncated && (obj_iter + 1)==objs.end()) {
-      list_op.params.marker = obj_iter->key;
-
-      int ret = fetch();
-      if (ret < 0) {
-        ldout(store->ctx(), 0) << "ERROR: list_op returned ret=" << ret << dendl;
-        return ret;
-      } 
-      obj_iter = objs.begin();
-      if (obj_iter == objs.end()) {
```

Comment:
What gets lost here is the check for obj_iter == objs.end() after calling fetch.  I'm not sure if this is possible?  If so, we should keep that check.

Reply:
In this context, fetch() will only be executed if the previous list_op is truncated, so if there were no errors in list_op, some entries would remain here. However in the meantime if the entire list diappear, there is theoretically a possibility that obj_iter is an end iterator.